tests/test_middleware_interface.py

```diff
@@ -145,26 +158,48 @@ def test_prep_butler(self):
         loaded_shards = list(self.butler.registry.queryDataIds("htm7",
                                                                datasets="gaia",
                                                                collections="refcats"))
+
+        # Check that the right skymap is in the chained output collection.
+        self.butler.datasetExists("skyMap",
+                                  # TODO: we shouldn't need skymap here?
+                                  skymap="deepCoadd_skyMap",
+                                  collections=self.interface.output_collection)
+
         # These shards were identified by plotting the objects in each shard
         # on-sky and overplotting the detector corners.
         # TODO DM-34112: check these shards again with some plots, once I've
         # determined whether ci_hits2015 actually has enough shards.
-        expected_shards = [157401, 157405, 157407]
+        expected_shards = [157394, 157401, 157405]
         self.assertEqual(expected_shards, [x['htm7'] for x in loaded_shards])
-        # check that we got appropriate calibs.
+        # Check that the right calibs are in the chained output collection.
         try:
             self.butler.datasetExists('cpBias', detector=56, instrument='DECam',
                                       collections="DECam/calib/20150218T000000Z")
+            # TODO: Have to use the exact calib collection, because we don't
+            # have validity ranges for data.
+            # collections=self.interface.output_collection)
         except LookupError:
             self.fail("Bias file missing from local butler.")
         try:
             self.butler.datasetExists('cpFlat', detector=56, instrument='DECam',
                                       physical_filter="g DECam SDSS c0001 4720.0 1520.0",
                                       collections="DECam/calib/20150218T000000Z")
+            # TODO: Have to use the exact calib collection, because we don't
+            # have validity ranges for data.
+            # collections=self.interface.output_collection)
         except LookupError:
             self.fail("Flat file missing from local butler.")
-        # TODO: check that we got a skymap
-        # TODO: check that we got a template
+
+        # Check that we configured the right pipeline.
+        self.assertEquals(self.interface.pipeline._pipelineIR.description,
+                          "End to end AP pipeline for the repo in /project/mrawls/hits2015-3")
+
+        # Check that the right templates are in the chained output collection.
+        for patch in (464, 465, 509, 510):
+            self.butler.datasetExists('deepCoadd', tract=0, patch=patch, band="g",
+                                      # TODO: we shouldn't need skymap here?
+                                      skymap="deepCoadd_skyMap",
```

Comment:
Is this a "just in case", or is the default inference actually not working?

Reply:
No, it is a requirement for whatever reason. Don't know why.

Reply:
Ok, [that's a problem](https://github.com/lsst-dm/prompt_prototype/blob/main/python/activator/activator.py#L89-L91). I can file a ticket to investigate.

Reply:
Yes, please file a ticket on this problem, and include some of the middleware folks in the cc:

Reply:
Filed as [DM-34241](https://jira.lsstcorp.org/browse/DM-34241).